### PR TITLE
Docs: Adds Webhook Limitation to EKS Install Doc

### LINKED ISSUE #29454: Docs: Add Webhook Limitations to EKS Install Doc
Currently, the EKS install [doc](https://docs.cilium.io/en/latest/installation/k8s-install-helm/) state the following limitations when using overlay mode:

>Cilium can alternatively run in EKS using an overlay mode that gives pods non-VPC-routable IPs. This allows running more pods per Kubernetes worker node than the ENI limit, but means that pod connectivity to resources outside the cluster (e.g., VMs in the VPC or AWS managed services) is masqueraded (i.e., SNAT) by Cilium to use the VPC IP address of the Kubernetes worker node.

Additionally, this mode does not support webhooks since the EKS control plane cannot directly access the pod. As a workaround, the webhook endpoint can be exposed using host networking. This workaround should be documented.

### PR DESCRIPTION
**Pull request by danehans**
For EKS installs that use overlay mode, webhook servers must be exposed outside the cluster to be reachable from the managed control plane.

Fixes #29454

### CODE REVIEW

File: Documentation/installation/k8s-install-helm.rst [hunk 1]
```diff
@@ -192,11 +192,18 @@ Install Cilium
...
-          the VPC IP address of the Kubernetes worker node. To set up Cilium 
-          overlay mode, follow the steps below:
+          Kubernetes worker node than the ENI limit but includes the following caveats:
+
+            1. Pod connectivity to resources outside the cluster (e.g., VMs in the VPC
+               or AWS managed services) is masqueraded (i.e., SNAT) by Cilium to use the
+               VPC IP address of the Kubernetes worker node.
+            2. `Admission webhooks <https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/#what-are-admission-webhooks>`_
+               must be reachable from outside the cluster since the EKS control plane runs
+               in a separate VPC from worker nodes. This can achieved by using host networking
```
Comment (ghost) [inline Documentation/installation/k8s-install-helm.rst]:
```suggestion
               must be reachable from the EKS control plane running in a separate VPC from worker nodes. This can achieved by using host networking
```
My understanding is that webhooks must be reachable by the control plane, but not anything else outside of the cluster, so I suggest to clarify it.

**Review by ghost — CHANGES_REQUESTED** [review]
Looks good, but I suggested one clarification.

Also requested a check from @cilium/aws

**Review by ghost — APPROVED** [review]
Thanks for the update, lgtm

File: Documentation/installation/k8s-install-helm.rst [hunk 2]
```diff
@@ -192,11 +192,17 @@ Install Cilium
...
-          connectivity to resources outside the cluster (e.g., VMs in the VPC
-          or AWS managed services) is masqueraded (i.e., SNAT) by Cilium to use
-          the VPC IP address of the Kubernetes worker node. To set up Cilium 
-          overlay mode, follow the steps below:
+          Kubernetes worker node than the ENI limit but includes the following caveats:
+
+            1. Pod connectivity to resources outside the cluster (e.g., VMs in the VPC
+               or AWS managed services) is masqueraded (i.e., SNAT) by Cilium to use the
+               VPC IP address of the Kubernetes worker node.
+            2. `Admission webhooks <https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/#what-are-admission-webhooks>`_
```
Comment (ungureanuvladvictor) [inline Documentation/installation/k8s-install-helm.rst]: I'd reword this to be more like: "The EKS API Server is unable to route packets to the overlay network. This implies that any webhook which needs to be accessed must be host networked or ...".

The end message I think we should convey is the following:
- there are multiple types of webhooks (e.g., validation, auditing) which are impacted by this
- the EKS API Server is not part of the overlay network -> is unable to communicate with that network

**Review by ungureanuvladvictor — CHANGES_REQUESTED** [review]

**Review by ungureanuvladvictor — APPROVED** [review]
LGTM

**Review by christarazi — APPROVED** [review]

**Comment by danehans** [comment]
Travis CI build failed due to:

```
Unable to find image 'quay.io/cilium/helm-toolbox:v1.0.1@sha256:a1483fc036b15c290326ba7ed36b8f0b714e185913b72a6074c04225d74e034c' locally
docker: Error response from daemon: Get https://quay.io/v2/: net/http: request canceled (Client.Timeout exceeded while awaiting headers).
```

File: Documentation/installation/k8s-install-helm.rst [hunk 3]
```diff
@@ -192,11 +192,17 @@ Install Cilium
...
-          connectivity to resources outside the cluster (e.g., VMs in the VPC
-          or AWS managed services) is masqueraded (i.e., SNAT) by Cilium to use
-          the VPC IP address of the Kubernetes worker node. To set up Cilium 
-          overlay mode, follow the steps below:
+          Kubernetes worker node than the ENI limit but includes the following caveats:
+
+            1. Pod connectivity to resources outside the cluster (e.g., VMs in the VPC
+               or AWS managed services) is masqueraded (i.e., SNAT) by Cilium to use the
+               VPC IP address of the Kubernetes worker node.
+            2. The EKS API Server is unable to route packets to the overlay network. This
```
Comment (joestringer) [inline Documentation/installation/k8s-install-helm.rst:200]: Minor nit while looking through afterwards, I think that unordered lists would be more appropriate here and would help to break up the flow of the document vs. the ordered list below. I don't think that there is any specific meaning to the order of these caveats, they're each unique points that the user may consider.
